src/script/engine.zig

```diff
@@ -476,29 +478,39 @@ pub const Engine = struct {
         std.debug.print("Attempt to execute invalid opcode: 0x{x:0>2}\n", .{self.script.data[self.pc]});
         return error.UnknownOpcode;
     }
-};
-
-test "Script execution - OP_1 OP_1 OP_EQUAL" {
-    const allocator = std.testing.allocator;
 
-    // Simple script: OP_1 OP_1 OP_EQUAL
-    const script_bytes = [_]u8{ 0x51, 0x51, 0x87 };
-    const script = Script.init(&script_bytes);
+    fn opSha1(self: *Engine) !void {
+        const data = try self.stack.pop();
+        var hash: [sha1.digest_length]u8 = undefined;
+        sha1.hash(data, &hash, .{});
+        try self.stack.pushByteArray(&hash);
+    }
+};
 
-    var engine = Engine.init(allocator, script, .{});
-    defer engine.deinit();
+// Testing SHA1 against known vectors
+test "sha1 vectors" {
+    const test_cases = [_]struct {
+        input: []const u8,
+        expected: []const u8,
+    }{
+        .{ .input = "hello", .expected = "aaf4c61ddcc5e8a2dabede0f3b482cd9aea9434d" },
+        .{ .input = "blockchain", .expected = "56fde8f4392113e0f19e0430f14502e06968669f" },
+        .{ .input = "abc", .expected = "a9993e364706816aba3e25717850c26c9cd0d89d" },
+        .{ .input = "bitcoin", .expected = "ed1b8d80793e70c0608e8a8508a8dd80f6aa56f9" },
+    };
+
+    for (test_cases) |case| {
+        errdefer {
+            std.log.err("test case failed, case = {s}", .{std.json.fmt(case, .{})});
+        }
+        var expected_output: [sha1.digest_length]u8 = undefined;
+        _ = try std.fmt.hexToBytes(&expected_output, case.expected);
 
-    try engine.execute();
+        var actual_output: [sha1.digest_length]u8 = undefined;
 
-    // Check if the execution result is true (non-empty stack with top element true)
-    {
-        const result = try engine.stack.pop();
-        defer allocator.free(result);
-        try std.testing.expect(result.len > 0 and result[0] != 0);
+        sha1.hash(case.input, &actual_output, .{});
```

Comment:
you are writing a test for sha1, which is part of the zig language std lib.
You should be writing a test for `opSha1`, the function you wrote

Reply:
Oh... will fix that too😁